Fix raising a different exception depending on platform when copying a Unix
socket using :func:`shutil.copyfile` by always raising a
:exc:`shutil.SpecialFileError` instead.
